packages/react-components/src/components/datatree-output-component.tsx

```diff
@@ -154,4 +160,45 @@ export class DataTreeOutputComponent extends AbstractOutputComponent<AbstractOut
         // fix Warning: Can't perform a React state update on an unmounted component
         this.setState = (_state, _callback) => undefined;
     }
+
+    protected async fetchSelectionData(): Promise<void> {
+        if (this.props.selectionRange) {
+            const origXYTree = cloneDeep(this.state.xyTree);
+            origXYTree.sort((a,b)=> (a.id > b.id) ? 1 : -1);
+
+            let payload: any;
+            if (Number(this.props.selectionRange.getStart()) < Number(this.props.selectionRange.getEnd())) {
+                payload = QueryHelper.timeQuery([this.props.selectionRange.getStart(), this.props.selectionRange.getEnd()]);
+            } else {
+                payload = QueryHelper.timeQuery([this.props.selectionRange.getEnd(), this.props.selectionRange.getStart()]);
+            }
+
+            payload.parameters.isFiltered = true;
+
+            // TODO: use the data tree endpoint instead of the xy tree endpoint
+            const tspClientResponse = await this.props.tspClient.fetchXYTree(this.props.traceId, this.props.outputDescriptor.id, payload);
+            const treeResponse = tspClientResponse.getModel();
+            if (tspClientResponse.isOk() && treeResponse) {
+                if (treeResponse.model) {
+                    this.setState({
+                        outputStatus: treeResponse.status,
+                        xyTree: treeResponse.model.entries,
+                    });
+                }
+            }
+        }
+    }
+
+    async componentDidUpdate(prevProps: DataTreeOutputProps): Promise<void> {
+        if (this.props.selectionRange && this.props.selectionRange !== prevProps.selectionRange) {
+            if (Math.abs(Number(this.props.selectionRange.getDuration())) > 0) {
```

Comment:
If the selection is changed to a single timestamp during the debounce period, the request will be sent with this single selection (start == end). So I initially thought that the duration check should be done at the end, inside the debounced method.

However, the resulting data is not empty statistics for the timestamp selection. After talking with Matthew, the stats include any examined segment that intersects with the selection. In Trace Compass the Selection statistics are shown even with a single timestamp selection. Once a selection has been done once, the Selection statistics will never disappear (because it's not possible for the user to unselect time). So I believe the duration check should just be removed and selection statistics should be fetched from the server regardless of whether the selection is a timestamp or a time range.